### DISCUSSION TITLE
IPXE wimboot error 0xc0000017

### DISCUSSION THREAD

**Discussion by grafalk** [ipxe/ipxe#1311] [Q&A]
HI All
We using IPXE and wimboot (last version) for load WINPE.
On Lenovo ThinkPad T14s Gen 4 we faced with strange error 0xc0000017 

![2024-09-28 13-59-02](https://github.com/user-attachments/assets/94073e4a-72bf-4a84-b87d-a4492a92e24a)

On the rest same model it's error not happen. 
After we try to update BIOS on the last version and it help. 
Of course we can update all our 1000 + laptops 
Of course, we can update the BIOS on all 1000 devices of this model, but maybe someone has faced  a similar problem ?

**Comment by NiKiZe** [comment]
How exactly does your iPXE script look, and could you share the P/N? (Might have one of these) T14s G4 covers a quite large range of machines ;)

But it is looking for `iaStorB.sys`
Have you modified your boot.wim in any way?

In regards to drivers, I would recommend to load network drivers separately, and then actually have the windows setup detect, and load disk drivers, that way you don't end up with weird incompatible or buggy drivers impacting devices that the drivers wasn't intended for.

The final recommendation if you don't want to follow my advice above, and do have a modified boot.wim, would be to update the drivers, or rebuild it from scratch with latest versions.

**Reply by grafalk** [reply]
Hi Christian

It's Lenovo T14s Gen 4 21F7 (https://pcsupport.lenovo.com/us/en/products/laptops-and-netbooks/thinkpad-t-series-laptops/thinkpad-t14s-gen-4-type-21f6-21f7/downloads)
Yes, boot.wim modified by SCCM (add HP winpe adrivers)
I'm too thought about drivers, but it's boot.wim fine working in wds and sccm and flash drive :) , and has issue only in ipxe.

regarding iPXE script I can provide it in monday

Thank you

**Reply by NiKiZe** [reply]
So you are using a modified win file with drivers for HP, which of course would work for HP, but who knows if that package contains all drivers for all versions, as I said I wouldn't include drivers in the wim at all, especially not disk drivers, and leave that up to the install process to search for and install instead.

If you don't want that path, then recreating the .wim with a working driver package (recommending intel official site for those) is the path you will have to take.
